File: Array/1441. Build an Array With Stack Operations/BuildArray.py
from typing import List

class Solution:
    def buildArray(self, target: List[int], n: int) -> List[str]:
        result = []
        index = 1
        for num in target:
            while num > index:
                result.append("Push")
                result.append("Pop")
                index += 1
            result.append("Push")
            index += 1
        return result

# 示例用法
solution = Solution()
target = [1, 3]
n = 3
print(solution.buildArray(target, n))

File: Array/1441. Build an Array With Stack Operations/BuildArray.cpp
#include <vector>
#include <string>

using namespace std;

class Solution {
public:
    vector<string> buildArray(vector<int>& target, int n) {
        vector<string> result;
        int index = 1;
        for (int i = 0; i < target.size(); i++) {
            while (target[i] > index) {
                result.push_back("Push");
                result.push_back("Pop");
                index++;
            }
            result.push_back("Push");
            index++;
        }
        return result;
    }
};